openapi-validation-core/src/test/java/com/getyourguide/openapi/validation/core/OpenApiRequestValidatorTest.java

```diff
@@ -1,40 +1,62 @@
 package com.getyourguide.openapi.validation.core;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
 
+import com.atlassian.oai.validator.model.SimpleRequest;
 import com.getyourguide.openapi.validation.api.metrics.MetricsReporter;
-import com.getyourguide.openapi.validation.api.model.ValidatorConfiguration;
+import com.getyourguide.openapi.validation.api.model.RequestMetaData;
+import com.getyourguide.openapi.validation.core.validator.OpenApiInteractionValidatorWrapper;
+import java.net.URI;
+import java.util.HashMap;
 import java.util.concurrent.RejectedExecutionException;
 import java.util.concurrent.ThreadPoolExecutor;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
 
 public class OpenApiRequestValidatorTest {
 
     private ThreadPoolExecutor threadPoolExecutor;
+    private OpenApiInteractionValidatorWrapper validator;
 
     private OpenApiRequestValidator openApiRequestValidator;
 
     @BeforeEach
     public void setup() {
         threadPoolExecutor = mock();
+        validator = mock();
         ValidationReportHandler validationReportHandler = mock();
         MetricsReporter metricsReporter = mock();
 
         openApiRequestValidator = new OpenApiRequestValidator(
             threadPoolExecutor,
             validationReportHandler,
             metricsReporter,
-            "",
-            new ValidatorConfiguration(null, null, null)
+            validator
         );
     }
 
     @Test
     public void testWhenThreadPoolExecutorRejectsExecutionThenItShouldNotThrow() {
-        Mockito.doThrow(new RejectedExecutionException()).when(threadPoolExecutor).execute(Mockito.any());
+        Mockito.doThrow(new RejectedExecutionException()).when(threadPoolExecutor).execute(any());
 
         openApiRequestValidator.validateRequestObjectAsync(mock(), null);
     }
+
+    @Test
+    public void testWhenEncodedQueryParamIsPassedThenValidationShouldHappenWithQueryParamDecoded() {
+        var uri = URI.create("https://api.example.com?ids=1%2C2%2C3");
```

Comment:
Any other cases we should test, e.g. when `&` or `=` is contained encoded in the value, or when we have multiple name-value pairs being passed?

Reply:
So, added more tests for these cases you mentioned (and also for spaces encoded with `+`). Hope happy you are 😉 .